### ISSUE TITLE
Operation node graph is missing the operation node [regression] 

### ISSUE THREAD

**Issue by vbergbauer** [kiali/kiali#3736]
Issue [#3644](https://github.com/kiali/kiali/issues/3644) is closed, so I am creating a new issue.

As of Kiali 1.30.0, the problem persists, with operation nodes disappearing from the graph.
I am using:
- Istio 1.9
- Kind 0.10.0 (Kubernetes 1.20.1)

You can see the issue with curl. Version 1.30.0:

```
curl -s "http://kiali.example/api/namespaces/graph?graphType=app&injectServiceNodes=true&namespaces=httpbin" | \
  jq '.elements.nodes[].data | {app, nodeType, aggregate}'
{
  "app": "httpbin",
  "nodeType": "box",
  "aggregate": null
}
{
  "app": "httpbin",
  "nodeType": "box",
  "aggregate": null
}
{
  "app": "httpbin",
  "nodeType": "app",
  "aggregate": null
}
{
  "app": "httpbin",
  "nodeType": "service",
  "aggregate": null
}
{
  "app": "istio-ingressgateway",
  "nodeType": "app",
  "aggregate": null
}
{
  "app": "httpbin",
  "nodeType": "app",
  "aggregate": null
}
{
  "app": "httpbin",
  "nodeType": "service",
  "aggregate": null
}
```

If I just replace the container image with 1.28.1 (leaving everything as it was), the service nodes appear again.

```bash
kubectl -n observability patch kiali observability-kiali --type=json \
  -p '[{"op": "replace", "path":"/spec/deployment/image_version", "value":"v1.28.1"}]'

while [[ $(kubectl -n observability get kiali observability-kiali -ojson | jq -r '.status.conditions[].reason') == "Running" ]]
do
  sleep 1
done
```

Version 1.28.1:

```
curl -s "http://kiali.example/api/namespaces/graph?graphType=app&injectServiceNodes=true&namespaces=httpbin" | \
  jq '.elements.nodes[].data | {app, nodeType, aggregate}'
{
  "app": "httpbin",
  "nodeType": "app",
  "aggregate": null
}
{
  "app": "httpbin",
  "nodeType": "aggregate",
  "aggregate": "request_operation=PostStatus"
}
{
  "app": "httpbin",
  "nodeType": "service",
  "aggregate": null
}
{
  "app": "httpbin",
  "nodeType": "aggregate",
  "aggregate": "request_operation=DeleteStatus"
}
{
  "app": "httpbin",
  "nodeType": "aggregate",
  "aggregate": "request_operation=GetStatus"
}
{
  "app": "istio-ingressgateway",
  "nodeType": "app",
  "aggregate": null
}
```

For reference, here are the specs of my Kiali CR, changing only the image version between tests:

```yaml
kind: Kiali
metadata:
  labels:
    app: observability-kiali
  name: observability-kiali
  namespace: observability
spec:
  auth:
    strategy: anonymous
  deployment:
    accessible_namespaces:
      - "**"
    image_version: v1.30.0
    ingress_enabled: false
    namespace: observability
    service_type: ClusterIP
    view_only_mode: false
  external_services:
    grafana:
      in_cluster_url: http://grafana:80
      url: http://grafana.example
    istio:
      component_status:
        enabled: false
    prometheus:
      in_cluster_url: http://prometheus-prometheus:9090
      url: http://prometheus-prometheus:9090
    tracing:
      in_cluster_url: http://jaeger-query:16686
      url: http://jaeger.example
  identity:
    cert_file: ""
    private_key_file: ""
  istio_component_namespaces:
    grafana: observability
    prometheus: observability
  istio_namespace: istio-system
  server:
    web_root: /
```

**Comment by lucasponce** [comment]
I remember there was a regression in Istio telemetry about this that yes, it may impact Kiali.
@jshaughn should have better details.

Thanks for opening the issue @vbergbauer and sorry for the delay.

**Comment by jshaughn** [comment]
@vbergbauer Would you mine testing against the `kiali-3673` image here: https://quay.io/repository/jshaughn/kiali?tab=tags ?

**Comment by vbergbauer** [comment]
Yes, the operations nodes show up now (and the TLS tags too).
Good:
![image](https://user-images.githubusercontent.com/8659572/110388589-e9884080-8028-11eb-98c8-c88ccf2c5324.png)

However...
Some graphs now see "double", with the service behind the gateway showing as an extra traffic source.
Not so good:
![image](https://user-images.githubusercontent.com/8659572/110389202-ce6a0080-8029-11eb-8eec-70ba14410061.png)

**Comment by jshaughn** [comment]
If you enable "cluster boxes" in the Display menu I think you may see that that entire section is inside the "unknown cluster".  If so then you may be seeing exactly what istio is reporting, you'd have to check it against what is in prometheus.  if those metrics are in prometheus then it may be an telemetry issue in istio.  Are you able to check prometheus, something like

istio_requests_total{source_cluster="unknown"}
istio_requests_total{destination_cluster="unknown"}

and see whether you see source and/or destination info for the bookinfo workloads in question?

**Comment by vbergbauer** [comment]
Hello @jshaughn 

- `istio_requests_total{source_cluster="unknown"}` returns nothing
- `istio_requests_total{destination_cluster="unknown"}` returns only the workloads without sidecars that I put behind my gateway (Kiali, Prometheus, Grafana) in the namespace `observability`.

So, no unexpected metrics there:
```
curl -s -G "http://prometheus.example/api/v1/query" \
  --data-urlencode query='istio_requests_total{destination_cluster="unknown"}' | \
  jq -r '.data.result[].metric.destination_service'
grafana.observability.svc.cluster.local
kiali.observability.svc.cluster.local
prometheus-prometheus.observability.svc.cluster.local
```

When I turn on then namespace and cluster boxes, this is what I see:
![image](https://user-images.githubusercontent.com/8659572/110472441-fb0e2e80-80a2-11eb-8fa4-c3c8e0e603cf.png)

**Comment by jshaughn** [comment]
I'll be honest, I can't quite explain why you see what you see.  Something does not add up.  The "unknown" cluster metrics must be coming from somewhere, In general Kiali is just reporting off of the Prom metrics.  You are sure you only have one Prom in play and are not maybe load-balancing queries between proms? (I know, it's a stretch question...)    I tried to DM you on istio slack so we could maybe go back and forth a bit.   

A couple more queries to play with, maybe.  I don't suppose you could expose your prometheus publically for me to look at?

sum(istio_requests_total{source_workload_namespace="bookinfo-content"} by (source_cluster, destination_cluster)
sum(istio_requests_total{destination_workload_namespace="bookinfo-content"} by (source_cluster, destination_cluster)

**Comment by vbergbauer** [comment]
Hi @jshaughn 
Istio 1.9.1 is indeed doing something strange.
I am running KinD on my laptop with one node, obviously single-cluster.
Still, I am getting different destination clusters in my queries, which makes no sense:

```
curl -s -G "http://prometheus.example/api/v1/query" --data-urlencode query='sum(istio_requests_total) by (source_cluster, destination_cluster)' | jq '.data.result'
[
  {
    "metric": {},
    "value": [
      1615338895.058,
      "23715"
    ]
  },
  {
    "metric": {
      "destination_cluster": "Kubernetes",
      "source_cluster": "Kubernetes"
    },
    "value": [
      1615338895.058,
      "5461"
    ]
  },
  {
    "metric": {
      "destination_cluster": "unknown",
      "source_cluster": "Kubernetes"
    },
    "value": [
      1615338895.058,
      "5596"
    ]
  }
]
```

**Comment by vbergbauer** [comment]
Hi @jshaughn, I got it.

My IstioOperator declared the extra dimension request_operation, like this:
```yaml
  values:
    telemetry:
      v2:
        prometheus:
          configOverride:
            inboundSidecar:
              metrics:
                - dimensions:
                    request_operation: istio_operationId
```

But then, I looked at https://github.com/istio/istio/blob/master/manifests/charts/istio-control/istio-discovery/templates/telemetryv2_1.9.yaml#L278 
I see that the override will remove `destination_cluster` and `source_cluster`.
So I need to re-add them, like this:

```yaml
  values:
    telemetry:
      v2:
        prometheus:
          configOverride:
            inboundSidecar:
              metrics:
                - dimensions:
                    request_operation: istio_operationId
                    destination_cluster: node.metadata['CLUSTER_ID']
                    source_cluster: downstream_peer.cluster_id
```
Then, everything works correctly.

**Comment by jshaughn** [comment]
Wow, @vbergbauer , GREAT find there.  This is definitely something to pursue on the Istio side, I wonder if they realize what's happening here.   I doubt it would be expected for folks to have to add back the cluster attributes.  [The docs ](https://istio.io/latest/docs/tasks/observability/metrics/classify-metrics/) for request classificatin don't really mention the operator, only how to manually update the stats_filter, so at a minimum there should probably be a doc update.

Normally I'd pursue this but since I'm off of work this week, I'd suggest you mention what you've found in the extension-telemetry channel on istio slack.  Or, if you don't mind waiting I can follow up next week.

**Comment by bianpengyuan** [comment]
We just removed cluster id customization at istio: https://github.com/istio/istio/pull/31226. So in 1.10, when you customize metrics, you on longer need to carry those cluster dimensions.

**Comment by jshaughn** [comment]
@bianpengyuan Ah, great to know, thanks, Peter!

**Comment by jshaughn** [comment]
I think this is resolved as of Istio 1.10, which is now released, so closing.  Thanks @vbergbauer.

**Comment by vbergbauer** [comment]
I verified it. It does work correctly on Istio 1.10
